lmcache/storage_backend/mem_pool/base_pool.py

```diff
@@ -0,0 +1,44 @@
+import abc
+from dataclasses import dataclass
+from typing import Optional
+
+import torch
+
+
+@dataclass
+class KVObj:
+    chunk_idx: int
+    data: torch.Tensor
+
+
+class BasePool(metaclass=abc.ABCMeta):
+    """
+    Interface for mem pool
+    """
+
+    @abc.abstractmethod
+    def allocate(self, kv_chunk: torch.Tensor) -> Optional[KVObj]:
+        """
+        Allocate a buffer memory pointer from the memory pool.
```

Comment:
Just to double check, the `kv_chunk` is only used to check the shape, right? We won't really copy the data into the allocated chunk.

Reply:
No real data copying is happening here:)